runtime/registry.go

```diff
@@ -81,29 +68,14 @@ func (r *Runtime) DeleteInstance(ctx context.Context, instanceID string, dropDB
 	return r.Registry().DeleteInstance(ctx, instanceID)
 }
 
-// EditInstance edits exisiting instance.
-// The API compares and only evicts caches if drivers or dsn is changed.
-// This is done to ensure that db handlers are not unnecessarily closed
+// EditInstance edits exisiting instance. Calling it will evict all connection handles associated with the instance.
 func (r *Runtime) EditInstance(ctx context.Context, inst *drivers.Instance) error {
-	_, err := r.connectorDef(inst, inst.OLAPConnector)
```

Comment:
Although we agreed to keep such checks in reconcile but it can still be useful to keep rudimentary API parameter checks.

Reply:
I thought about it, but there's very little to check here – in theory, neither are required, and also, the OLAP connector could be defined in `rill.yaml` (so checks would fail here). In practice of course, we always set them to more or less hard-coded values.

Reply:
I actually had this issue in mind: https://github.com/rilldata/rill/pull/3066
Without this check it would have broken an existing running project. 

>the OLAP connector could be defined in rill.yaml

But while creating instance we would always define a project connector right ?

Reply:
It's a valid point, but worst case is that people see a "connector not found" error message on the project logs (at least won't fail silently). 

Yes in practice we currently hard-code them, but in theory we can support setting the OLAP connector to `myduckdb` and then only defining it in `rill.yaml`, like:

```
# rill.yaml
...
connectors:
- name: myduckdb
  type: duckdb
  dsn: path/to/my/duck.db
```

Additionally, there's a fair amount of functionality that can work with no OLAP or repo configured, like connector preflight, and ad-hoc reconcile resources (like the upcoming scheduled reports, which will be added to the catalog through API requests instead of from code files). 

Not needing to configure them can also be useful for testing.